File: Aitech/AITechC2/app/src/main/java/com/vn/tcshop/aitechc/Models/CartHistory.java
package com.vn.tcshop.aitechc.Models;

public class CartHistory {
    private final String code_cart;
    private final int cart_status;
    private final String cart_date;
    private final int cart_number;
    private final double cart_cost;
    private final String cart_payment;

    public CartHistory(String code_cart, int cart_status, String cart_date, int cart_number, double cart_cost, String cart_payment) {
        this.code_cart = code_cart;
        this.cart_status = cart_status;
        this.cart_date = cart_date;
        this.cart_number = cart_number;
        this.cart_cost = cart_cost;
        this.cart_payment = cart_payment;
    }

    public String getCode_cart() {
        return code_cart;
    }

    public int getCart_status() {
        return cart_status;
    }

    public String getCart_date() {
        return cart_date;
    }

    public int getCart_number() {
        return cart_number;
    }

    public double getCart_cost() {
        return cart_cost;
    }

    public String getCart_payment() {
        return cart_payment;
    }
}

File: Aitech/AITechC2_PHP/includes/CartHistory.php
<?php

class CartHistory{
    private $con;
    private $image_url = "http://192.168.1.12:86/AITechC2/quanlysanpham/uploads/";

    function __construct() {
        //Getting the DbConnect.php file
        require_once dirname(__FILE__) . '/DbConnect.php';

        //Creating a DbConnect object to connect to the database
        $db = new DbConnect();

        //Initializing our connection link of this class
        //by calling the method connect of DbConnect class
        $this->con = $db->connect();
    }

    function getCartHistory($id_user){
        $stmt = $this->con->prepare("SELECT cart_payments.code_cart, cart_payments.cart_status, cart_payments.cart_date, cart_payments.cart_number, cart_payments.cart_cost, cart_payments.cart_payment FROM cart_payments, shipping, users WHERE users.id = shipping.id_user AND shipping.id_shipping = cart_payments.id_shipping AND users.id = ? ORDER BY cart_payments.id_cart DESC");
        $stmt->bind_param('i', $id_user);
        $stmt->execute();
        $stmt->bind_result($code_cart, $cart_status, $cart_date, $cart_number, $cart_cost, $cart_payment);

        $cart_histories = array();

        while ($stmt->fetch()){
            $cart_history = new stdClass();
            $cart_history->code_cart = $code_cart;
            $cart_history->cart_status = $cart_status;
            $cart_history->cart_date = $cart_date;
            $cart_history->cart_number = $cart_number;
            $cart_history->cart_cost = $cart_cost;
            $cart_history->cart_payment = $cart_payment;

            array_push($cart_histories, $cart_history);
        }
        return $cart_histories;
    }
}

?>